enclave_build/src/pull.rs

```diff
@@ -0,0 +1,388 @@
+                        )
+                    })?
+                    .to_string();
+
+                let auth = auth.replace('"', "");
+                // Decode the auth token
+                let decoded = base64::decode(&auth).map_err(|err| {
+                    EnclaveBuildError::CredentialsError(format!(
+                        "Invalid Base64 encoding for auth: {}",
+                        err
+                    ))
+                })?;
+                let decoded = std::str::from_utf8(&decoded).map_err(|err| {
+                    EnclaveBuildError::CredentialsError(format!(
+                        "Invalid utf8 encoding for auth: {}",
+                        err
+                    ))
+                })?;
+
+                // Try to get the username and the password
+                if let Some(index) = decoded.rfind(':') {
+                    let (username, after_user) = decoded.split_at(index);
+                    let (_, password) = after_user.split_at(1);
+
+                    return Ok(RegistryAuth::Basic(
+                        username.to_string(),
+                        password.to_string(),
+                    ));
+                }
+            }
+        }
+    }
+
+    // If the auth token is missing, return error
+    Err(EnclaveBuildError::CredentialsError(
+        "Credentials not found.".to_string(),
+    ))
+}
+
+/// Determines the authentication for interacting with the remote registry.
+fn registry_auth(image_name: &str) -> RegistryAuth {
+    match parse_credentials(image_name) {
+        Ok(registry_auth) => {
+            let (user, pass) = if let RegistryAuth::Basic(user, pass) = registry_auth {
+                (user, pass)
+            } else {
+                (String::new(), String::new())
+            };
+            RegistryAuth::Basic(user, pass)
+        }
+        Err(err) => {
+            eprintln!("Credentials error: {:?}, performing anonymous pull.", err);
+            RegistryAuth::Anonymous
+        }
+    }
+}
+
+/// Pulls an image (all blobs - layers, manifest and config) from a Docker remote registry.
+pub async fn pull_image_data(image_name: &str) -> Result<ImageData> {
+    // Build the client required for the pulling - uses HTTPS protocol
+    let mut client = build_client(ClientProtocol::Https);
+
+    let image_ref = image::build_image_reference(image_name)
+        .map_err(|err| EnclaveBuildError::ImagePullError(err.to_string()))?;
+
+    let auth = registry_auth(image_name);
+
+    // Pull from remote an ImageData struct containing the layers, manifest and configuration files
+    client
+        .pull(&image_ref, &auth, ACCEPTED_MEDIA_TYPES.to_vec())
+        .await
+        .map_err(|err| EnclaveBuildError::ImagePullError(err.to_string()))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        io::{Read, Write},
+        path::PathBuf,
+    };
+
+    use super::*;
+    use serde_json::{json, Value};
+    use sha2::Digest;
+    use tempfile::{tempdir, TempDir};
```

Comment:
I know this is used in other places in this project as well, but it's an external dependency and we should try to avoid them when that's now too difficult to do. Can you please use [tempfile](https://github.com/rust-vmm/vmm-sys-util/blob/main/src/tempfile.rs) from vmm-sys-util? It shouldn't be too hard to switch to that.

We should also open an issue, to remove the tempfile dependency from the whole project.